src/com/google/guiceberry/junit4/GuiceBerryTestRule.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.guiceberry.junit4;
+
+import com.google.guiceberry.DefaultEnvSelector;
+import com.google.guiceberry.GuiceBerry;
+import com.google.guiceberry.GuiceBerryEnvSelector;
+import com.google.guiceberry.TestDescription;
+import com.google.inject.Module;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * {@link GuiceBerry} adapter for JUnit4 tests, implementing {@TestRule}.
+ *
+ * @author David M. Hull
+ */
+public class GuiceBerryTestRule extends GuiceBerryBaseRule implements TestRule {
+  private final Object target;
+
+  private GuiceBerryTestRule(Object target, GuiceBerryEnvSelector guiceBerryEnvSelector) {
+    super(guiceBerryEnvSelector);
+    this.target = target;
+  }
+
+  public GuiceBerryTestRule (Object target, Class<? extends Module> envClass) {
+    this(target, DefaultEnvSelector.of(envClass));
+  }
+
+  @Override
```

Comment:
for better or worse, GuiceBerry is set of java 1.5 source code compatibility, where @Override is illegal for impls of interface methods.

Strangely, the ant system (build.xml + build.properties) is set to declare source and target to be 1.5, and, yet, they are not barfing at this. I spent some time looking into it and I can't understand what is going on -- everything seems to be configured correctly, and, yet, it seems to happily compile... The compilation process does say:

"source value 1.5 is obsolete and will be removed in a future release"

which is making me think that this is just a bug in Oracle's java compiler, against a neglected piece of code... :( It is also making me think it might be time to bring GuiceBerry to 1.6, though I would probably bump the version to 5.0.0. Considering that Guice has, itself, upgraded to 1.6 (I just checked), I'm quite inclined to make this change.

On related news, the tutorial is not compiled by ant (which is, arguably, a bug), but it should also remain compliant to 1.5. But your code has some 1.8 constructs.

Would you please fix all these 1.5 violations?